flutter_app/lib/modules/get_drone_information.dart

```diff
@@ -1,48 +1,97 @@
 import 'dart:async';
+import 'dart:typed_data';
+import 'package:dart_mavlink/mavlink.dart';
+import 'package:dart_mavlink/dialects/common.dart';
 import 'package:imacs/modules/mavlink_communication.dart';
 
 class GetDroneInformation {
+  final MavlinkParser _parser = MavlinkParser(MavlinkDialectCommon());
   final MavlinkCommunication comm;
 
-  GetDroneInformation({required this.comm});
+  final StreamController<double> _yawStreamController =
+      StreamController<double>();
+  final StreamController<double> _pitchStreamController =
+      StreamController<double>();
+  final StreamController<double> _rollStreamController =
+      StreamController<double>();
+  final StreamController<double> _rollSpeedController =
+      StreamController<double>();
+  final StreamController<double> _pitchSpeedController =
+      StreamController<double>();
+  final StreamController<double> _yawSpeedController =
+      StreamController<double>();
+  final StreamController<int> _timeBootMsPitchController =
+      StreamController<int>();
+
+  final StreamController<int> _latStreamController = StreamController<int>();
+  final StreamController<int> _lonStreamController = StreamController<int>();
+  final StreamController<int> _altStreamController = StreamController<int>();
+
+  GetDroneInformation({required this.comm}) {
+    _parseMavlinkMessage();
+    comm.stream.listen((Uint8List data) {
```

Comment:
Instead of doing this, we can make `_tcpSocket` public (or add getter to it) and listen to it that way, somehow that fixes the bug